### DISCUSSION TITLE
Connect WebGPU to Syphon/Spout

### DISCUSSION THREAD

**Discussion by h-a-n-n-e-s** [gpuweb/gpuweb#4494] [Ideas/Feedback]
[Syphon](https://syphon.github.io) (Mac) and [Spout](https://spout.zeal.co) (Windows) are technologies to share data (frames, textures, ...) directly on the GPU between different apps, with essentially zero latency. As the links above show, there is already a large number of apps, mainly from the gaming/art/visualization communities (Blender, Unity, Touchdesigner,...) which use it.

I think it would be great if a WebGPU app could use Syphon/Spout because then it could interface easily with all these other apps, opening up many new creative possibilities.

There are SDK's for [Syphon](https://github.com/Syphon/Syphon-Framework/blob/main/Syphon.docc/GettingStarted.md) and [Spout](https://github.com/leadedge/Spout2), but I'm not sure how to set up such an interface to which WebGPU could connect to.

From a [discussion](https://github.com/leadedge/Spout2/issues/111#issuecomment-1953409841) with people developing Spout there are some comments like:
> The application would need to find the rendering procedure and from that get an OpenGL texture or Fbo for the sending functions. The texture cannot be mipmapped. It can have only one level.

which might help.

**If anybody has an idea how this interface could look like and how to set it up, please get in touch.**

**Comment by shaoboyan** [comment]
I don't think this is **WebGPU Spec related** issue.
And I think to achieve this means we need to "get" the surface from browser. In multiple-process architecture browser, e.g. Chrome, it means we need to "get" resource from GPU process and send it to another library. I think this might have security concerns from browser perspective, but just some personal thoughts.

**Reply by kainino0x** [reply]
Agreed, this isn't really WebGPU related - it could apply to any image data in a browser. It's more of a browser implementation proposal. In Chrome, it's very unlikely that Google would implement an integration, but if a Chromium contributor made it work for Chromium embedders (like Electron) then it's possible it could be accepted into Chromium. You could ask on graphics-dev@chromium.org, it sounds like it would relate to some of the window-capture/tab-capture/casting paths in Chromium today.
